sql/moz-fx-data-shared-prod/telemetry_derived/retention_clients_v1/query.sql

```diff
@@ -0,0 +1,139 @@
+WITH clients_last_seen AS (
+  SELECT
+    cls.submission_date,
+    cls.client_id,
+    cls.sample_id,
+    cls.normalized_channel,
+    mozfun.bits28.retention(cls.days_seen_bits, cls.submission_date) AS retention_seen,
+    mozfun.bits28.retention(
+      cls.days_active_bits & cls.days_seen_bits,
+      cls.submission_date
+    ) AS retention_active,
+    cls.days_seen_bits,
+    cls.days_active_bits,
+  FROM
+    `moz-fx-data-shared-prod.telemetry.clients_last_seen_v2` cls
+  WHERE
+    cls.submission_date = @submission_date
+),
+new_profiles AS (
+  SELECT
+    cfs.client_id,
+    cfs.sample_id,
+    first_seen_date,
+    country,
+    locale,
+    app_version,
+    attribution_campaign,
+    attribution_content,
+    attribution_dlsource,
+    attribution_medium,
+    attribution_ua,
+    distribution_id,
+    -- cfs.isp_name,
+    cfs.normalized_channel,
+    startup_profile_selection_reason,
+    attribution_experiment,
+    -- mozfun.norm.os(cfs.os) AS normalized_os,
+    cfs.normalized_os,
+    -- COALESCE(
+    --   mozfun.norm.windows_version_info(cfs.os, cfs.os_version, cfs.windows_build_number),
+    --   NULLIF(SPLIT(cfs.normalized_os_version, ".")[SAFE_OFFSET(0)], "")
+    -- ) AS normalized_os_version,
+    cfs.normalized_os_version,
+    submission_date,
```

Comment:
since this is from the LEFT JOIN table and we don't want it to be NULL on the final table I suggest we use
`COALESCE(cls.submission_date, DATE_ADD(cfs.first_seen_date, INTERVAL 27 day)) as submission_date,`

Reply:
updated